lib/PhpParser/Token.php

```
@@ -8,11 +8,14 @@
 class Token extends Internal\TokenPolyfill {
     /** Get (exclusive) zero-based end position of the token. */
     public function getEndPos(): int {
-        return $this->pos + \strlen($this->text);
+        return $this->pos < 0 ? $this->pos : $this->pos + \strlen($this->text);
```

Hm, I wonder if it wouldn't be better to throw an exception instead, if no end position/line is available.

---

@nikic i'm lazy) this case need rewrites many tests (: